### DISCUSSION TITLE
https://theme-hope.vuejs.press/zh/guide/interface/icon.html

### DISCUSSION THREAD

**Discussion by giscus** [vuepress-theme-hope/giscus-discussions#606] [Announcements]
# https://theme-hope.vuejs.press/zh/guide/interface/icon.html

整个主题在多处都添加了 FontClass / 图片 格式图标的支持。 目前你可以使用 iconfont、iconify 和 fontawesome 为你的项目添加图标，可以设置自己的图标资源。 同时，png/svg 格式的图标也是支持的。你可以使用完整 URL 或路径名来添加图标。 我们推荐使用 iconify 或 fontawesome。

https://theme-hope.vuejs.press/zh/guide/interface/icon.html

**Comment by 1226085293** [comment]


**Reply by Mister-Hope** [reply]
如果你只使用 iconfont 图标资源的话，我们会自动为你添加 iconfont 类。

https://github.com/vuepress-theme-hope/vuepress-theme-hope/blob/c4bb5f6191c33df8a4b7d517004ebc1566ab4093/packages/components/src/node/define.ts#L31

https://github.com/vuepress-theme-hope/vuepress-theme-hope/blob/c4bb5f6191c33df8a4b7d517004ebc1566ab4093/packages/components/src/node/components/fontIcon.ts#L14

https://github.com/vuepress-theme-hope/vuepress-theme-hope/blob/c4bb5f6191c33df8a4b7d517004ebc1566ab4093/packages/components/src/node/components/fontIcon.ts#L52

反之，我们不会，也就是你应该使用诸如 `iconfont icon-home` `fab fa-discover` 之类的类名来显式指定一个图表系列。

**Reply by Mister-Hope** [reply]
我是真笑了，本来我们会自动把iconPrefix设置为iconfont，然后你自己把它改了，然后跑过来评论，说我们图标类名错了。

**Reply by 1226085293** [reply]
知道了，之前没仔细看文档

**Comment by 1226085293** [comment]
并且混用也不行，只有 iconAssets[0] 生效
`
		iconAssets: [
			"//at.alicdn.com/t/c/font_4195463_l6p35g6g8a9.css",
			"iconfont",
		],
`

**Reply by Mister-Hope** [reply]
我们会为你注入每一个图标资源，但不同浏览器对重复的字体定义的处理不同。如果你使用的浏览器不支持自动合并多个字体声明，则的确可能出现第一个或最后一个生效。

此外，不同项目的不同图标可能占用相同的私有字符。所以我们在任何情况下都不鼓励你引用多个 iconfont 链接。

**Comment by theCodedMike** [comment]
Hello，问一下，为什么有的精选icon很快就能显示，有的就不行了呢？比如home这个icon很快就能显示，而support这个就无法正常显示？
还有，从iconfont下载的svg放在public文件夹下，此时再使用这个icon后，页面中的icon显得很不和谐，这个是为什么呢？

**Reply by Mister-Hope** [reply]
很快？iconfont现在的链接是有限速的，仅供调试，建议自托管文件，fontawsome 是更好的方案。

**Reply by zhuonixian** [reply]
Hello   有解决吗

**Comment by spots11** [comment]
那么我有一个png图标放在public下，该如何在侧边栏使用呢？

**Reply by Mister-Hope** [reply]
直接填入绝对路径

**Reply by spots11** [reply]
这样？
{
  icon: '图标绝对路径'
  text: ''
 .......
}
我这样渲染出来是<span class="font_icon icon 绝对路径">，我后面是直接改了依赖包渲染的结果，改成<img src=''>才显示出来

**Comment by GodX-18** [comment]
内部支持的图标列表怎么没有了

**Reply by Mister-Hope** [reply]
移除了，如果你需要可以看一下历史提交中的地址。我们鼓励使用fontawesome，以避免产生版权问题

**Reply by mwmzuishuai** [reply]
https://fontawesome.com/search

**Comment by theCodedMike** [comment]
没解决，我后来就只用能正常显示的icon。

我看了下，现在"图标支持"这个页面变化了很多，你看看现在有没有好的解决方案，我很久没关注这里了。
________________________________
发件人: zhuonixian ***@***.***>
发送时间: 2024年6月7日 23:02
收件人: vuepress-theme-hope/giscus-discussions ***@***.***>
抄送: theCodedMike ***@***.***>; Comment ***@***.***>
主题: Re: [vuepress-theme-hope/giscus-discussions] https://theme-hope.vuejs.press/zh/guide/interface/icon.html (Discussion #606)


Hello 有解决吗

―

**Comment by zhuonixian** [comment]
fontawesome 上的icon在线的 现在不知道为啥  有时候就不显示了  起初以为是server的网络问题到 公网资源。后面确认的网络也是通的，自己下载的SVG   没有在线那种不通位置显示大小或者深色 浅色的效果，很别扭哈...

**Reply by Mister-Hope** [reply]
1. 一个非常基础的事情是图标是客户端也就是用户浏览器获取的，与用户网络相关，这个稍微有点网页调试基础的人都能看得出来。不要扯什么服务器访问没问题。

2. 自己下载的svg需要自己调整图标在viewbox的显示相对位置至统一。比如对于两个viewbox均为1024*1024的图标，一个四周有200的留白，一个完全没有那显然不会渲染成相同大小。

3. 主题自托管了 fontawesome，以避免国内jsdeliver不稳定的问题，你也可以自托管，详见fontawsome文档。这些并不是应该塞在主题文档里的东西。

**Reply by zhuonixian** [reply]
感谢回答。 是jsdeliver不稳定的原因，看fontawesome也是走的jsdeliver。有些走js加载超时了
https://cdn.jsdelivr.net/npm/@fortawesome/fontawesome-free@6/js/fontawesome.min.js
...

**Comment by sola-grj** [comment]
请问下，font-awesome在hope2.0中是什么版本的呢

**Reply by Mister-Hope** [reply]
latest v6

**Comment by haotengda** [comment]
请教下，fontawesome图标库是访问cdn.jsdelivr.net加载的，要怎么配置从自己服务器中加载，我是在内网部署。将iconAssets配置成“/js/fontawesome.min.js”，虽然从自己服务器中加载到浏览器本地了，但是图标不展示。

**Reply by Mister-Hope** [reply]
需要手动设置iconPrefix

**Comment by xiaojincaigithub** [comment]
我想知道，怎么设置背景图，我没找到哪里改，小白一个

**Comment by theCodedMike** [comment]
你直接问一下这个主题的原作者吧，给他发邮件或者加QQ群什么的。我看了一下我的项目，我现在也忘记当时是怎么加的了，感觉这个主题的官网有变化。

________________________________
发件人: 善 ***@***.***>
发送时间: 2024年12月29日 22:39
收件人: vuepress-theme-hope/giscus-discussions ***@***.***>
抄送: theCodedMike ***@***.***>; Comment ***@***.***>
主题: Re: [vuepress-theme-hope/giscus-discussions] https://theme-hope.vuejs.press/zh/guide/interface/icon.html (Discussion #606)


我想知道，怎么设置背景图，我没找到哪里改，小白一个

―

**Comment by leoleor** [comment]
我想知道页面上的图标要怎么设置，我看作者推荐用FontAwesome，我是这样写icon: b:apple，但是没有出现，以前的精选图标不知道在哪看的

**Reply by zhuonixian** [reply]
icon走的 cf cdn 国内不是太稳定
